File: leibniz.go
package main

import "fmt"
import "math"
import "os"
import "strconv"

func LeibnizPow(i int) float64 {
	return math.Pow(-1, float64(i)) / (2 * float64(i) + 1)
}

func Leibniz(n int) float64  {
	r := 0.0
	for i := 0; i < n; i++ {
		r += LeibnizPow(i)
	}
	return 4 * r
}

func main() {
	if c, err := strconv.Atoi(os.Args[1]); err == nil {
		res := Leibniz(c)
		fmt.Printf("%f", res)
	}
}

File: leibniz.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>

double leibnizPow(int i) {
  return pow(-1, i) / (2 * i + 1);
}

double leibniz(int n) {
  double r = .0;
  for (int i = 0; i < n; i++) {
    r += leibnizPow(i);
  }
  return 4 * r;
}

int main(int argc, char *args[]) {
  int c = atoi(args[1]);
  double p = leibniz(c);
  printf("%1.9f \n", p);
  return 0;
}
